### PR TITLE
[LLM] Speculative decoding HF llama example for CPU

### PR DESCRIPTION
**Pull request by qiyuangong**
## Description

LLM Speculative Decoding example for CPU

### 1. Why the change?

Speculative Decoding can speed up generation with an assistant model. In this example, the assistant model is an int4 version of the main model, so we don't have to load two different models.

### 4. How to test?
- [ ] N/A
- [ ] Unit test
- [X] Application test
- [X] Document test
- [ ] ...

### CODE REVIEW

File: python/llm/example/CPU/HF-Transformers-AutoModels/Model/llama2/README.md [hunk 1]
```diff
@@ -84,3 +84,27 @@ What is AI?
...
+
+```bash
+python speculative_decoding.py --repo-id-or-model-path REPO_ID_OR_MODEL_PATH --assistant-model-path ASSISTANT_MODEL_PATH --prompt PROMPT --n-predict N_PREDICT
+```
+
+New Arguments info:
+- `--assistant-model-path ASSISTANT_MODEL_PATH`: argument defining the huggingface repo id for the assistant model (e.g. `meta-llama/Llama-2-7b-chat-hf` and `meta-llama/Llama-2-13b-chat-hf`) to be downloaded, or the path to the huggingface checkpoint folder. 
+
+#### 3.1 Use additional model as assistant model
+By default, we need to add a second model named draft/assistant model to generate draft proposals. Note that assistant model should be similar to main model. It's recommended to use a small version of main model, e.g., Llama-72B as main model and Llama-7B as assistant model.
```
Comment (jason-dai) [inline python/llm/example/CPU/HF-Transformers-AutoModels/Model/llama2/README.md:99]: there is no llama-72B model

Comment (qiyuangong) [reply python/llm/example/CPU/HF-Transformers-AutoModels/Model/llama2/README.md]: Changed to 70b

File: python/llm/example/CPU/HF-Transformers-AutoModels/Model/llama2/README.md [hunk 2]
```diff
@@ -84,3 +84,27 @@ What is AI?
...
+
+```bash
+python speculative_decoding.py --repo-id-or-model-path REPO_ID_OR_MODEL_PATH --assistant-model-path ASSISTANT_MODEL_PATH --prompt PROMPT --n-predict N_PREDICT
+```
+
+New Arguments info:
+- `--assistant-model-path ASSISTANT_MODEL_PATH`: argument defining the huggingface repo id for the assistant model (e.g. `meta-llama/Llama-2-7b-chat-hf` and `meta-llama/Llama-2-13b-chat-hf`) to be downloaded, or the path to the huggingface checkpoint folder. 
+
+#### 3.1 Use additional model as assistant model
+By default, we need to add a second model named draft/assistant model to generate draft proposals. Note that assistant model should be similar to main model. It's recommended to use a small version of main model, e.g., Llama-70B as main model and Llama-7B as assistant model.
```
Comment (jason-dai) [inline python/llm/example/CPU/HF-Transformers-AutoModels/Model/llama2/README.md:99]: llama2-70b and llama2-7b

Comment (qiyuangong) [reply python/llm/example/CPU/HF-Transformers-AutoModels/Model/llama2/README.md]: Addressed.

**Pull request closed without merging**